spec/aruba/platform/windows_environment_vars_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'spec_helper'
+
+# create a value that does not exist in the collection. append a number to the original_str until not found
+def nonexistant(coll = [], str = "zzz")
+  not_found_value = str
+  i = 1
+  while coll.include?(not_found_value)
+    not_found_value = "#{not_found_value}-#{i}"
+    i += 1
+  end
+  not_found_value
+end
+
+describe WindowsEnvironmentVars do
+  subject(:windows_env) { described_class.new }
+
+  let(:env_1)         { "EnvVarAdapt-1" }
+  let(:env_1_value)   { "EVA one one one" }
+  let(:env_2)         { "EnvVarAdapt-2" }
+  let(:env_2_value)   { "EVA two two two" }
+  let(:default_value) { "default value" }
+
+  it "doesn't have any lower-case letters" do
+    expect(windows_env.to_h.select {|k, v| (k =~ /[a-z]+/) }).to be_empty
+  end
+
+  describe "only finds upper-case keys" do
+    before(:each) do
+      @orig_env = ENV.to_hash
```

Comment:
Let's not fiddle around with `ENV`. The logic in `Windows|UnixEnvironmentVariables` is independent of `ENV`.